### PR TITLE
integration: refactor, cleanup, and add more tests for TokenRequest

### PR DESCRIPTION
**Pull request by mikedanese**
ref #58790

```release-note
NONE
```

### CODE REVIEW

File: test/integration/auth/svcaccttoken_test.go [hunk 1]
```diff
@@ -63,103 +64,162 @@ func TestServiceAccountTokenCreate(t *testing.T) {
...
-		Spec: v1.PodSpec{
-			ServiceAccountName: sa.Name,
-			Containers:         []v1.Container{{Name: "test-container", Image: "nginx"}},
-		},
-	}
-	_, err = cs.CoreV1().Pods(pod.Namespace).Create(pod)
-	if err != nil {
-		t.Fatalf("err: %v", err)
-	}
+		if treq, err := cs.CoreV1().ServiceAccounts(sa.Namespace).CreateToken(sa.Name, treq); err == nil {
```
Comment (cjcullen) [inline test/integration/auth/svcaccttoken_test.go]: Call the response `tres` or `tresp` or something.

Comment (cjcullen) [reply test/integration/auth/svcaccttoken_test.go]: I know they're the same type, and shadowing prevents you from overwriting the first one, but it confuses my small brain.

Comment (mikedanese) [reply test/integration/auth/svcaccttoken_test.go]: Fixed

**Pull request merged**